daemon/cluster/listen_addr.go

```diff
@@ -0,0 +1,159 @@
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+func resolveListenAddr(specifiedInterface, specifiedAddr string) (string, error) {
+	specifiedHost, specifiedPort, err := net.SplitHostPort(specifiedAddr)
+	if err != nil {
+		return specifiedAddr, nil
+	}
+	ip := net.ParseIP(specifiedHost)
+	if ip != nil && ip.IsUnspecified() {
+		// The user has not provided a listen address.
+		if specifiedInterface != "" {
+			interfaceAddr, err := resolveInterfaceAddr(specifiedInterface)
+			if err != nil {
+				return "", err
+			}
+			return net.JoinHostPort(interfaceAddr.String(), specifiedPort), nil
+		}
+
+		systemAddr, err := resolveSystemAddr()
+		if err != nil {
+			return "", err
+		}
+		return net.JoinHostPort(systemAddr.String(), specifiedPort), nil
+	} else if specifiedInterface != "" {
+		return "", errors.New("cannot specify both --listen-interface and --listen-addr")
```

Comment:
Oh, wait we have just `--listen-address` and `--listen-interface` now of course. Yeah, I agree, sorry about the noise; I saw your suggestion for three separate flags. If we don't expect more options to be added, that is